### PR TITLE
feat(adapter): firebase adapter

### PR DESCRIPTION
**Pull request by leifermendez**
Firebase adapter:

Qwik should be in everything ecosystem, so I bring a new adapter for Firebase. I still working to improve the dx

```pnpm run qwik add firebase```

demo: https://qwik-firebase-demo.web.app/

- [x] Feature / enhancement
- [ ] Bug
- [ ] Docs / tests / types / typos

# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

# Use cases and why

- 1. One use case
- 2. Another use case

# Checklist:

- [ ] My code follows the [developer guidelines of this project](https://github.com/BuilderIO/qwik/blob/main/CONTRIBUTING.md)
- [ ] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] Added new tests to cover the fix / functionality

### CODE REVIEW

**Comment by alexcsandru** [comment]
Hello Qwik developers and leifermendez,

I hope this message finds you all well. I'm really passionate about the adapter for Firebase hosting feature as it's something that could greatly benefit our Qwik community.

Though I have been trying my best, my technical knowledge on this particular feature is somewhat limited and I've been facing a few hurdles. Is there anyone available who's experienced in this area and could potentially guide us to  complete this feature?

Eagerly looking forward to any updates or input. Thank you all in advance for your time, expertise and continuous collaboration. Let's keep pushing boundaries!

**Comment by leifermendez** [comment]
> Hello Qwik developers and leifermendez,
> 
> I hope this message finds you all well. I'm really passionate about the adapter for Firebase hosting feature as it's something that could greatly benefit our Qwik community.
> 
> Though I have been trying my best, my technical knowledge on this particular feature is somewhat limited and I've been facing a few hurdles. Is there anyone available who's experienced in this area and could potentially guide us to complete this feature?
> 
> Eagerly looking forward to any updates or input. Thank you all in advance for your time, expertise and continuous collaboration. Let's keep pushing boundaries!

HI @alexcsandru currently this feature is waiting to merge. So I hope to soon be available for everyone :)

**Comment by bezael** [comment]
Hello, everyone.
Is there any forecast on when we will have this feature?

Thanks!

PS: good job @leifermendez

**Comment by mhevery** [comment]
Sorry about the daly, Let me see if we can get this merged....

**Comment by alexcsandru** [comment]
This will be awesome! Thanks Miško Hevery @mhevery

File: starters/adapters/firebase/package.json [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    "deploy": "firebase deploy"
+  },
+  "devDependencies": {
+    "firebase-tools": "12.4.3"
+  },
+  "__qwik__": {
+    "priority": 30,
+    "displayName": "Adapter: Firebase",
+    "docs": [
+      "https://qwik.builder.io/qwikcity/adaptors/aws-lambda/"
```
Comment (Mouvedia) [inline starters/adapters/firebase/package.json:15]: Is it the right url?

**Pull request merged**